### PR TITLE
fix: Close tty first in `ptytest` cleanup

### PR DESCRIPTION
**Pull request by mafredri**
This is an attempt to make ptytest closure even more robust, due to https://github.com/coder/coder/issues/5247.

### CODE REVIEW

File: pty/ptytest/ptytest.go [hunk 1]
```diff
@@ -72,11 +72,29 @@ func create(t *testing.T, ptty pty.PTY, name string) *PTY {
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
 		defer cancel()
 
-		_ = out.Close()
-		_ = ptty.Close()
```
Comment (mtojek) [inline pty/ptytest/ptytest.go:76]: Of course the submitted solution is more robust, but I'm wondering if it doesn't help if you just change the order here, so:

```
ptty.Close()
out.Close()
```

File: pty/ptytest/ptytest.go [hunk 2]
```diff
@@ -72,11 +72,29 @@ func create(t *testing.T, ptty pty.PTY, name string) *PTY {
...
+		errC := make(chan error, 2)
```
Comment (mtojek) [inline pty/ptytest/ptytest.go:75]: nit: should `errC` be closed?

**Review by mtojek — APPROVED** [review]

File: pty/ptytest/ptytest.go (hunk 1 above)
Comment (mafredri) [reply pty/ptytest/ptytest.go:76]: My hunch is that if we're running to an error here, it's `ptty.Close()` that hangs. Only changing the order _could_ help, but if it doesn't then we would never hit the timeout (if this is the error).

File: pty/ptytest/ptytest.go (hunk 2 above)
Comment (mafredri) [reply pty/ptytest/ptytest.go:75]: Nope, we're waiting for `copyDone`, closing `errC` would exit cleanup too early.

File: pty/ptytest/ptytest.go (hunk 1 above)
Comment (mafredri) [reply pty/ptytest/ptytest.go:76]: Well, let's go for the simple approach first. I've updated the PR, what do you think?

Comment (mtojek) [reply pty/ptytest/ptytest.go:76]: 👍 

Ship it!

**Pull request merged**